phoenix-core-client/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -470,6 +471,7 @@ private IndexMaintainer(final PTable dataTable, final PTable cdcTable, final PTa
         this.immutableStorageScheme = index.getImmutableStorageScheme() == null ? ImmutableStorageScheme.ONE_CELL_PER_COLUMN : index.getImmutableStorageScheme();
         this.dataEncodingScheme = dataTable.getEncodingScheme() == null ? QualifierEncodingScheme.NON_ENCODED_QUALIFIERS : dataTable.getEncodingScheme();
         this.dataImmutableStorageScheme = dataTable.getImmutableStorageScheme() == null ? ImmutableStorageScheme.ONE_CELL_PER_COLUMN : dataTable.getImmutableStorageScheme();
+        this.nDataTableSaltBuckets = isDataTableSalted ? dataTable.getBucketNum() : PTable.NO_SALTING;
```

Comment:
Any reason why are you introducing the special value of -1 ? The field `nIndexSaltBuckets` is set to 0 if index is not salted. We should maintain consistency.

Reply:
In the existing code, `PTableImpl.toProto()` translates `0` into `-1` and `PTableImpl.createFromProto()` leaves this value as `null` by checking for `-1`. To be consistent, I think we should change `nIndexSaltBuckets` to also `-1` instead of the other way around. I did a quick check, I think the existing code will work as is, as it checks for `> 0`. WDYT?

Reply:
Sounds good @haridsv